velox/type/Filter.h

```diff
@@ -865,6 +868,100 @@ class BigintValuesUsingHashTable final : public Filter {
         values_(other.values_),
         sizeMask_(other.sizeMask_) {}
 
+  T min() const {
+    return min_;
+  }
+
+  T max() const {
+    return max_;
+  }
+
+  const std::vector<T>& values() const {
+    return values_;
+  }
+
+  const std::vector<T>& hashTable() const {
+    return hashTable_;
+  }
+
+  bool testInt(T value) const {
+    auto kEmptyMarker =
+        std::is_same_v<T, int128_t> ? kHugeintEmptyMarker : kBigintEmptyMarker;
+    auto kMurmurHash =
+        std::is_same_v<T, int128_t> ? kHugeintMurmurHash : kBigintMurmurHash;
+
+    if (containsEmptyMarker_ && value == kEmptyMarker) {
+      return true;
+    }
+    if (value < min_ || value > max_) {
+      return false;
+    }
+    uint32_t pos = (value * kMurmurHash) & sizeMask_;
+    for (auto i = pos; i <= pos + sizeMask_; i++) {
+      int32_t idx = i & sizeMask_;
+      T l = hashTable_[idx];
+      if (l == kEmptyMarker) {
+        return false;
+      }
+      if (l == value) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+ protected:
+  const T min_;
+  const T max_;
+  std::vector<T> hashTable_;
+  bool containsEmptyMarker_ = false;
+  std::vector<T> values_;
+  int32_t sizeMask_;
+
+  // Empty marker for the int64_t hash table.
+  static constexpr int64_t kBigintEmptyMarker = 0xdeadbeefbadefeedL;
```

Comment:
Can these be moved to the corresponding sub-classes?

Reply:
Thanks for the suggestion.. but because the hashtable construction and insertion of values from the inList for both ```BigintValuesUsingHashTable``` and ```HugeintValuesUsingHashTable``` is happening in the base class, ```IntValuesUsingHashTable```, it cannot be moved without duplicating this code.